fmralign/metrics.py

```diff
@@ -0,0 +1,180 @@
+    elif loss is "corr":
+        score = np.array([pearsonr(X_gt[:, vox], X_pred[:, vox])[0]  # pearsonr returns both rho and p
+                          for vox in range(X_pred.shape[1])])
+    else:
+        raise NameError(
+            "Unknown loss. Recognized values are 'R2', 'corr', or 'reconstruction_err'")
+    # if the calculated score is less than -1, return -1
+    return np.maximum(score, -1)
+
+
+def normalized_reconstruction_error(y_true, y_pred, sample_weights=None,
+                                    multioutput='raw_values'):
+    """
+    Calculates the normalized reconstruction error
+    as defined by Bazeille and colleagues (2019).
+    
+    A perfect prediction yields a value of 1.
+    
+    Parameters
+    ----------
+    y_true : arr
+        The ground truth array.
+    y_pred : arr
+        The predicted array.
+    sample_weights : arr
+        Weights to assign to each sample.
+    multioutput: str in ['raw_values', 'uniform_average',
+                         'variance_weighted']
+        Defines aggregating of multiple output scores.
+        Default is raw values.
+        'raw_values' :
+            Returns a full set of scores in case of multioutput input.
+        'uniform_average' :
+            Scores of all outputs are averaged with uniform weight.
+        'variance_weighted' :
+            Scores of all outputs are weighted by the variance of a given
+            sample.
+    
+    Returns
+    -------
+    score : float or ndarray of floats
+        The score or ndarray of scores if `multioutput` is `raw_values`.
+
+    References
+    ----------
+    `Bazeille T., Richard H., Janati H., and Thirion B. (2019) Local
+    Optimal Transport for Functional Brain Template Estimation.
+    In: Chung A., Gee J., Yushkevich P., and Bao S. (eds) Information 
+    Processing in Medical Imaging. Lecture Notes in Computer Science,
+    vol 11492. Springer, Cham.
+    DOI: 10.1007/978-3-030-20351-1_18.`
+    """
+    if y_true.ndim == 1:
+        y_true = y_true.reshape((-1, 1))
+
+    if y_pred.ndim == 1:
+        y_pred = y_pred.reshape((-1, 1))
+
+    numerator = ((y_true - y_pred) ** 2).sum(axis=0, dtype=np.float64)
+    denominator = ((y_true) ** 2).sum(axis=0, dtype=np.float64)
+
+    # Include only non-zero values
+    nonzero_denominator = (denominator != 0)
+    nonzero_numerator = (numerator != 0)
+    valid_score = (nonzero_denominator & nonzero_numerator)
+
+    # Calculate reconstruction error
+    output_scores = np.ones([y_true.shape[-1]])
+    output_scores[valid_score] = 1 - (numerator[valid_score] /
+                                      denominator[valid_score])
+    if multioutput == 'raw_values':
+        # return scores individually
+        return output_scores
+    
+    elif multioutput == 'uniform_average':
+        # passing None as weights yields uniform average
+        return np.average(output_scores, weights=None)
+
+    elif multioutput == 'variance_weighted':
```

Comment:
Unclear to me what the use case for that is.

Reply:
see above

Reply:
OK. It's always a bad  to have some unused / ill-motivated features, but i have no better idea.